marimo/_cli/cli.py

```diff
@@ -184,9 +184,11 @@ def edit(
     name: Optional[str] = None,
 ) -> None:
     if name is not None:
-        temp_dir = tempfile.TemporaryDirectory()
         # Validate name, or download from URL
-        name = validate_name(name, allow_new_file=True, temp_dir=temp_dir)
+        # The second return value is an optional temporary directory. It is
```

Comment:
It turns out `delete=True/False` is only available in Python 3.12. In older versions the lifetime of the directory on disk is bound to the object (similar to the shared memory object), so I just changed `validate_name` to return the temporary directory to the caller, who keeps it around.